src/udev/udev-node.c

```diff
@@ -204,7 +204,7 @@ static int stack_directory_find_prioritized_devnode(sd_device *dev, const char *
                         return -ENOMEM;
         }
 
-        dir = opendir(dirname);
+        dir = opendir(FORMAT_PROC_FD_PATH(dirfd));
```

Comment:
typically `xopendirat(dirfd, ".", O_NOFOLLOW)` is preferable, since it won't require /proc/ then.

Reply:
nice, thanks for the suggestion.